### DISCUSSION TITLE
Get children documents with API

### DISCUSSION THREAD

**Discussion by glungtung** [outline/outline#6187] [General] [answered]
Hi,
What should I use from API to get a document children documents, ideally with a shareId (without auth) ?
Something like a ".children" field in data answer when retrieving a document with /documents.info API call, is it possible ?
How are you achieving this with the public view page of a shared document ?
Thanks !

**Comment by tommoor** [comment, marked as answer]
If you just want the next level of children you can query `documents.list` by `parentDocumentId`.

If you want the entire tree you can either use `collections.documents` which will give you the entire tree for the collection, or `documents.info` with `apiVersion=2` will include the tree below the given document in the response.

**Comment by glungtung** [comment]
Hi, thanks ! 
`apiVersion=2` is exactly what I need,
is this API V2 already documented somewhere ?

**Reply by tommoor** [reply]
It's not, it's supposed to be internal-only and actually we plan to [change it to a header soon ](https://github.com/outline/outline/pull/5553/files#diff-9144cf8ce9cc765774a8ef9c433dc48ef416bb043dede912f1e4908674e6dd6eR1451)– so keep an eye out for that change if you decide to go this route.

**Reply by glungtung** [reply]
Ok thanks a lot
 I'll go this way because it allows me to have children list without authentification
I will keep an eye on future upgrades if there is a change with this API call, do you plan to integrate this in next release ?

**Reply by tommoor** [reply]
I plan to, both options will work for at least one release

**Reply by glungtung** [reply]
When using `documents.info` with `apiVersion=2` which gives me the tree of shared children, how could I get children content ?
API answer only gives me "id", "url" and "title", but I would need "shareId" to get content, just as the public view page does. 
Is there a way to get children content with parent shareId + "url" of child maybe ?

**Reply by tommoor** [reply]
You can use the same `shareId` to load the child documents, take a look at the requests made on https://docs.getoutline.com/s/guide as an example.

**Reply by glungtung** [reply]
Nice example thanks
I still don't find how to get children content through API
I can have the `sharedTree` and get for example 
            {
              "id": "402edeca-d45c-4e08-8eeb-533d304d4b0f",
              "url": "/doc/webhooks-gB7HYhS6yq",
              "title": "Webhooks",
              "children": []
            }
But now how could I get "Webhooks" page content through API ?
I've tried to call `documents.info` with `"shareId": "guide/doc/webhooks-gB7HYhS6yq"` but it doesn't work : `"message": "shareId: Invalid input"`

**Reply by tommoor** [reply]
`shareId` is just "guide" in this example, typically it's a UUID.

**Reply by glungtung** [reply]
`documents.info` with `"shareId": "guide", "apiVersion": "2"` is working perfectly ❤️
I get first page content, and sharedTree with children's id, url and title

❓But how do i get children's page content ?

**Reply by glungtung** [reply]
I got this : 
```
    "sharedTree": {
      "id": "f9c52a3c-bccb-4e74-a1a4-cb02b51bb1a7",
      "url": "/doc/guide-7dWGlDrtL7",
      "title": "Guide",
      "children": [
        {
          "id": "aad82868-6259-4d5a-aead-800c6f2b4719",
          "url": "/doc/getting-started-CB6Ipl9Cdc",
          "title": "Getting started",
          "children": [
            {
              "id": "6fee1047-2b9e-4474-aeee-f9578e0d2eb4",
              "url": "/doc/the-app-Ez4bmY6VDD",
              "title": "The app",
              "children": []
            },
```

But i don't know how to get "Getting started" content or "The app" content
I guess it's possible because public view page can do it, so it should be accessible through API
I'd be thankfull for some little more help

**Reply by tommoor** [reply]
Also pass to `documents.info` the `id` of the nested document with the `shareId` you already have.

**Reply by glungtung** [reply]
Ok i got it, thanks !
